### PR TITLE
Add additional paramerters to chip tool darwin

### LINKED ISSUE #17148: TE9 Chip-tool-Darwin Multifabric testing: Optional parameters of [--timedInteractionTimeoutMs] [--fabric-filtered] needs to be supported
In all of the Multifabric tests (ECM and BCM), in chip-tool-darwin, support is needed for optional parameters,  [--timedInteractionTimeoutMs] [--fabric-filtered]
Chip-tool currently supports these optional parameters but chip-tool-darwin doesn't. Below is the image of chip-tool vs chip-tool-darwin
Screen Shot 2022-04-06 at 1 39 33 PM
Screen Shot 2022-04-06 at 2 50 58 PM

### PR DESCRIPTION
**Pull request by krypton36**
#### Problem
* Adds the ability to provide fabricFiltered, timedInteractionTimeoutMs and timedWriteTimeoutMs in chip-tool-darwin CLI.
* Fixes #17148

#### Change overview
- Adds fabricFiltered, timedInteractionTimeoutMs and timedWriteTimeoutMs to zap 
- Generated code

#### Testing
- Generated code
- Compiled
- Ran commands with parameters.

### CODE REVIEW

File: examples/chip-tool-darwin/templates/commands.zapt [hunk 1]
```diff
@@ -211,6 +218,7 @@ public:
         AddArgument("attr-name", "{{asDelimitedCommand (asUpperCamelCase name)}}");
         AddArgument("min-interval", 0, UINT16_MAX, &mMinInterval);
         AddArgument("max-interval", 0, UINT16_MAX, &mMaxInterval);
+        AddArgument("fabric-filtered", 0, 1, &mFabricFiltered);
```
Comment (tcarmelveilleux) [inline examples/chip-tool-darwin/templates/commands.zapt:221]: Why add this to Darwin only, and not Linux as well?

This seems like this should not be Darwin-only

Comment (bzbarsky-apple) [reply examples/chip-tool-darwin/templates/commands.zapt:221]: `chip-tool` (which runs on both Linux and Darwin and uses the SDK C++ API directly) already supports these.

`chip-tool-darwin`, which runs on Darwin only and uses the Darwin framework API, not the SDK directly is what this PR is about.

File: examples/chip-tool-darwin/templates/commands.zapt [hunk 2]
```diff
@@ -155,6 +159,7 @@ public:
         {{else}}
         AddArgument("attr-value", {{asTypeMinValue type}}, {{asTypeMaxValue type}}, &mValue);
         {{/if_chip_complex}}
+        AddArgument("timedWriteTimeoutMs", 0, UINT16_MAX, &mTimedWriteTimeoutMs);
```
Comment (bzbarsky-apple) [inline examples/chip-tool-darwin/templates/commands.zapt:162]: This is not API-compatible with `chip-tool`, which uses `timedInteractionTimeoutMs` for writes.  Is that purposeful, or an oversight?

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/chip-tool-darwin/templates/commands.zapt (hunk 1 above)
Comment (tcarmelveilleux) [reply examples/chip-tool-darwin/templates/commands.zapt:221]: This is really confusing. These tools should be better named to express what they are more specifically

Comment (bzbarsky-apple) [reply examples/chip-tool-darwin/templates/commands.zapt:221]: Would renaming chip-tool-darwin to darwin-matter-tool help?

There's a tradeoff here between executable name length and expressiveness...

File: examples/chip-tool-darwin/templates/commands.zapt (hunk 2 above)
Comment (krypton36) [reply examples/chip-tool-darwin/templates/commands.zapt:162]: This was definitely an oversight. Thank you for the catch. I will update it to be timedInteractionTimeoutMs.

Comment (krypton36) [reply examples/chip-tool-darwin/templates/commands.zapt]: Updated.

File: examples/chip-tool-darwin/templates/commands.zapt (hunk 1 above)
Comment (krypton36) [reply examples/chip-tool-darwin/templates/commands.zapt:221]: @tcarmelveilleux  We have an [issue](https://github.com/project-chip/connectedhomeip/issues/14987) tracking the rename of this CLI. We haven't officially decided on a name.

**Comment by bzbarsky-apple** [comment]
fast-tracking platform-specific test tool change.

**Pull request merged**